File: algorithms/Binary Number with Alternating Bits/Binary Number with Alternating Bits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        bin_n = bin(n)[2:]
        for i in range(len(bin_n)-1):
            if bin_n[i] == '0' and bin_n[i+1] == '0':
                return False
            
            if bin_n[i] == '1' and bin_n[i+1] == '1':
                return False
            
        return True

File: algorithms/B/Binary Number with Alternating Bits/Binary Number with Alternating Bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        String l = Integer.toBinaryString(n);
        for(int i = 0 ; i< l.length() -1 ; i++){
            if(l.charAt(i) == l.charAt(i+1))
                return false;
        }
        return true;
    }
}
